File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/shared/StandardRequest.java
package com.lasthopesoftware.bluewater.shared;

import org.slf4j.LoggerFactory;

import java.io.InputStream;
import java.util.HashMap;

import javax.xml.parsers.SAXParser;
import javax.xml.parsers.SAXParserFactory;

public class StandardRequest {

	private final boolean status;
	public final HashMap<String, String> items = new HashMap<String, String>();
	
	public StandardRequest(String status) {
		this.status = status != null && status.equalsIgnoreCase("OK");
	}

	/**
	 * @return the status
	 */
	public boolean isStatus() {
		return status;
	}
	
	public static StandardRequest fromInputStream(InputStream is) {
		try {
			final SAXParser sp = SAXParserFactory.newInstance().newSAXParser();
	    	final StandardResponseHandler jrResponseHandler = new StandardResponseHandler();
			sp.parse(is, jrResponseHandler);
			return jrResponseHandler.getResponse();
		} catch (Exception e) {
			LoggerFactory.getLogger(StandardRequest.class).error(e.toString(), e);
		}
    	
    	return null;
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/shared/StandardRequest.kt
package com.lasthopesoftware.bluewater.shared

import org.slf4j.LoggerFactory
import java.io.InputStream
import javax.xml.parsers.SAXParserFactory

class StandardRequest internal constructor(status: String?) {
    /**
     * @return the status
     */
    val isStatus: Boolean
    val items = HashMap<String?, String?>()

    init {
        isStatus = status != null && status.equals("OK", ignoreCase = true)
    }

    companion object {
        fun fromInputStream(`is`: InputStream?): StandardRequest? {
            try {
                val sp = SAXParserFactory.newInstance().newSAXParser()
                val jrResponseHandler = StandardResponseHandler()
                sp.parse(`is`, jrResponseHandler)
                return jrResponseHandler.response
            } catch (e: Exception) {
                LoggerFactory.getLogger(StandardRequest::class.java).error(e.toString(), e)
            }
            return null
        }
    }
}
